src/pipdeptree/_models/dag.py

```diff
@@ -164,7 +152,7 @@ def filter_nodes(self, include: list[str] | None, exclude: set[str] | None) -> P
                             # a dependency is missing
                             continue
 
-        non_existent_includes = [i for i in include_with_casing_preserved if i.lower() not in matched_includes]
+        non_existent_includes = [i for i in include if i not in matched_includes]
```

Comment:
I don't think we should change this as elements in `include` and `exclude` comes straight from user input and we should normalize what they give us (but now that I see it, I think we should do better and perform PEP 503 normalization here on the strings).